SleepEngine/Engine/Components/Component.h

```diff
@@ -4,16 +4,28 @@
 
 BEGIN_NAMESPACE_SLEEP
 
+class Transform2D;
+class Object;
+
 class Component
 {
     FORBID_COPY_AND_MOVE(Component)
 public:
     Component(TypeID id)
         : m_typeID(id)
     {}
+    virtual ~Component() {}
+
+    virtual void update(float dt) {}
+    void setParent(Object* parent) 
+    { 
+        m_parent = parent; 
+        onAddedAsParent();
+    }
+
+    Transform2D& getTransform();
+    Transform2D const& getTransform() const;
```

Comment:
Please, don't put such methods in Component. Let's create `Transform2D getComponent(Object& objectWithTransform)` global function.